File: packages/backend/src/features/game/game.model.ts
import { Column, Entity } from 'typeorm';
import { QuizzemModel } from '~/src/core/domain/quizzem.model';
import { type GameState } from '~/src/features/game/game-state.model';

@Entity()
export class GameModel extends QuizzemModel {
  @Column()
  state: GameState = 'DRAFT';
}

File: packages/backend/src/main/kotlin/quizzem/backend/features/game/model/GameModel.kt
package quizzem.backend.features.game.model

import jakarta.persistence.*
import quizzem.backend.core.domain.QuizzemModel
import quizzem.backend.features.question.model.QuestionModel

@Entity
@Table(name = "games")
class GameModel(
    @Enumerated(EnumType.STRING)
    var state: GameState = GameState.DRAFT,

    @Column(length = 100)
    var title: String,

    @Column(columnDefinition = "TEXT")
    var description: String? = null,

    @ManyToMany(fetch = FetchType.LAZY)
    @JoinTable(
        name = "games_questions",
        joinColumns = [JoinColumn(name = "games_id")],
        inverseJoinColumns = [JoinColumn(name = "questions_id")]
    )
    var questions: MutableList<QuestionModel> = mutableListOf(),
) : QuizzemModel()